webapp/src/app.tsx

```diff
@@ -4038,6 +4038,14 @@ export class ProjectView
             });
     }
 
+    async getBlocksAsync(): Promise<Blockly.Block[]> {
+        if (!this.isBlocksActive()) {
+            await this.openBlocksAsync();
```

Comment:
I wonder if we should just throw an error in this case rather than switch to blocks. Having that switch be a side-effect of another call feels like it could lead to some unwanted behavior...plus, if we throw an error instead of switching, this function doesn't have to be async :)

Reply:
Good point!